CIPs/0018.md

```diff
@@ -0,0 +1,53 @@
+# CIP [0018]: Change group epoch score calculation for voter rewards
+
+- Date: 2020-08-30
+- Author: @zviadm
+- Status: DRAFT
+
+## Overview
+
+Currently group epoch score is calculated as an average score of its members epoch score
+for voter rewards. 
+
+Because of this, larger groups with more validators in them have a distinct advantage for 
+voters since their average score per epoch is likely going to be higher since they are able to 
+average out individual bad validator scores with better performing validators.
+
+This seems counter productive since for network health it would be better to incentivize voters
+to vote for smaller groups for more decentralization.
+
+Fix for this can be very simple, we change group epoch score to be:
+* Minimum(member validator epoch scores), instead of Avg(member validator epoch scores).
+
+Calculating score per epoch using minimum incetinvizes voting for smaller groups since they
+are more likely to maintain better epoch scores, and it also puts more pressure on larger groups
+to make sure all their validators maintain high uptime.
+
+This is a pretty simple change, and it only needs to be changed in one place here:
+calculateGroupEpochScore function in Validators.sol
+https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/governance/Validators.sol#L418
+
+## Goals
+
+- Shift voter incentive to vote for smaller groups instead of larger groups.
+- Provide more incentive for larger groups to maintain high uptime for all of their validators.
+
+## Proposed Solution
+
+
+Change calculateGroupEpochScore function in Validators.sol to calculate group score using 
+a minimum instead of average.
+https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/governance/Validators.sol#L418
+
+## Alternative Solutions
+
+## Risks
+
```

Comment:
One risk I see here is that it makes adding a new validator, one without an established score, to the group an expensive operation. In the epochs immediately following the addition of the new validator, the score of the group of be limited by the growth of that validator's score. This is already the case to a lesser extent, but this change would increase the penalty that the group sees on it's rewards.

Reply:
Validator epoch score used in Group epoch score calculations doesn't contain history (i.e. its not the cumulative validator score). It only contains score for that single epoch so this shouldn't be an issue.

Reply:
Ah, right. I got mixed up 👍